### PR TITLE
Some mobile improvements

### PR DESCRIPTION
**Pull request by hamishtaplin**
Screenshot 2021-04-22 at 13 27 03

### What

Made improvements to design on small screens:

- add mobile nav
- move config of nav sections to the base template
- change base font-size on small screens to `16px`

### How to review

Check that the mobile nav is fit for purpose:

- a11y
- focus/hover states (focus _styling_ is not considered yet, we should probably do this as a separate piece across the app)
- general usability

### Who can review

@armstrongb @will-madebymoon  @rav-pradhan @dandwal

### CODE REVIEW

File: src/views/layouts/default.njk [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+        pages: collections.components | eleventyNavigation
+    }
+] %}
+
 <!DOCTYPE html>
-<html lang="en">
+<html lang="en" class="text-grey-1">
     <head>
         <meta charset="utf-8">
         <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
Comment (dandwal) [inline src/views/layouts/default.njk:22]: Not part of this PR but I haven't looked at the other PRs until this point. Not a requirement for this PR but the meta tag 
```html
<meta description="{{ description }}">
```
seems strange to me.
Shouldn't it be something like
```html
<meta name="description" content="{{ description }}"
```
Unless I am missing the use case of it.

File: src/img/icons/chevron-down.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
```
[the remark this answers is not in the document]
Comment (dandwal) [reply src/img/icons/chevron-down.svg:1]: If we wanted a new home for other image assets in this repo to live under a new directory at the root I can add a new directory for terraform to apply along with updating the security group for CI to deploy there when the cdn.ons.gov.uk-assets changes.

File: src/views/partials/header.njk [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
 
-<header class="mb-4 bg-night-blue text-white">
-    <div class="max-w-7xl mx-auto">
-        <div class="py-4 px-6 text-lg font-bold">Digital Publishing Design System</div>
+<header class="bg-ocean-blue text-white">
+    <div class="page-container flex items-center justify-between py-3">
+        <a href="/" class="sm:text-lg font-bold leading-5 no-underline text-white hover:text-white">Digital Publishing Design System</a>
+        <button id="btn-menu" class="ml-9 py-2 pl-2 pr-1 font-bold text-sm md:hidden border-white border flex items-center relative active:top-px" aria-expanded="false">
+            <span class="uppercase">Menu</span>
+            <svg id="btn-menu-arrow" class="transform" width="20" height="20" viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
```
Comment (dandwal) [inline src/views/partials/header.njk:10]: Might be worth replacing chevron-down.svg with this and then referencing that here instead
(can wrap it around an `<object>` tag)

File: src/views/partials/mobile-nav.njk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                {% if section.title %}
+                    <button class="text-base border-b border-grey-4 hover:bg-grey-4 text-left w-full text-grey-1 px-4 p-3 pr-2 flex items-center justify-between nav-mobile-btn" data-index="{{loop.index0}}" >{{section.title}}
+                        <svg class="transform -rotate-90" width="20" height="20" viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg" aria-hidden="true"><path d="M5 6H15L10 15L5 6Z"/></svg>
+                    </button>
+                {% endif %}
+                <ul class="md:mb-4 {% if section.title %} hidden {% endif %}" id="nav-mobile-section-{{loop.index0}}">
+                    {% for post in section.pages %}
+                        {% set isActive = page.url == post.url %}
+                        <li class="border-b border-grey-4">
+                            <a href="{{post.url | url }}" class="block px-4 py-3 no-underline hover:bg-grey-4 border-ocean-blue border-opacity-0 border-l-4 border-inset  {% if isActive %} border-opacity-100  {% endif %} " {% if isActive %} aria-current="page" {% endif %}>
```
Comment (dandwal) [inline src/views/partials/mobile-nav.njk:14]: Nothing major just an observation some files have spaces on the inside of the variable lookups `{{ something }}` (rather than `{{something}}` think typical njk is to have it the spaces

File: src/views/partials/nav.njk [hunk 5]
```diff
@@ -1,31 +1,14 @@
...
     <ul>
         {% for section in sections %}
             <li>
                 {% if section.title %}
                     <h2 class="font-normal text-base text-grey-2 mb-2">{{section.title}}</h2>
                 {% endif %}
-                <ul class="mb-4">
+                <ul class="md:mb-4">
                     {% for post in section.pages %}
                         {% if page.url == post.url %}
```
Comment (dandwal) [inline src/views/partials/nav.njk:10]: I know not being introduced here but the tabbing in this file is a little funky, else and endif don't align.

File: src/views/partials/nav.njk [hunk 6]
```diff
@@ -1,31 +1,14 @@
...
-                            <li class="mb-2 pl-2 -ml-3 font-bold border-l-4 border-ocean-blue" aria-current="page">
+                            <li class="mb-2 pl-2 md:-ml-3 font-bold border-l-4 border-ocean-blue" aria-current="page">
```
Comment (dandwal) [inline src/views/partials/nav.njk:11]: Should this <li> be closed there are three opening ones now and only two closing

**Review by dandwal — COMMENTED** [review]
Looks really good great work. The only comment stopping me hit approve is the extra <li> tag with no closing one. The rest of the comments are either not to do with what this PR introduces or are minor

File: src/views/layouts/default.njk (hunk 1 above)
Comment (hamishtaplin) [reply src/views/layouts/default.njk:22]: Yep, you're right—well spotted. I'll change it here.

File: src/views/partials/header.njk (hunk 3 above)
Comment (hamishtaplin) [reply src/views/partials/header.njk:10]: Yeah this is a good point - I forgot to finish this but off but ideally it would be the same sag but also take advantage of `currentColor`. Will see what I can do.

File: src/views/partials/mobile-nav.njk (hunk 4 above)
Comment (hamishtaplin) [reply src/views/partials/mobile-nav.njk:14]: I think there's more instances without spaces than with them. I'm surprised Prettier isn't picking this up (I have "format on save" turned on).

File: src/views/partials/nav.njk (hunk 5 above)
Comment (hamishtaplin) [reply src/views/partials/nav.njk:10]: Yeah this is the fault of Prettier (or whatever it is formatting .njk files in my vscode) - I'll take a look at something more robust.

File: src/views/partials/nav.njk (hunk 6 above)
Comment (hamishtaplin) [reply src/views/partials/nav.njk:11]: I know it looks weird but the logic is:

```
{% if page.url == post.url %}
    <li class="mb-2 pl-2 md:-ml-3 font-bold border-l-4 border-ocean-blue" aria-current="page">
{% else %}
    <li class="mb-2">
{% endif %}
        <a href="{{post.url | url }}" class="no-underline hover:underline">{{post.title}}</a>
    </li>
{% endfor %}
```

So there's only ever one opening `<li>` due to the if/else. It's not the cleanest way of doing it anyway, I'll see if I can spread the attributes or something instead.

File: src/views/partials/nav.njk (hunk 5 above)
Comment (hamishtaplin) [reply src/views/partials/nav.njk:10]: Looking at it, I think the if/else combined with `<li>` is what's confusing it.

File: src/img/icons/chevron-down.svg (hunk 2 above)
[the remark this answers is not in the document]
Comment (hamishtaplin) [reply src/img/icons/chevron-down.svg]: I'm leaving for now as these are tiny icons that aren't used anywhere else so I think inlining them (and the benefits of `currentColor` inlining brings) means they are ok

File: src/views/partials/header.njk (hunk 3 above)
Comment (hamishtaplin) [reply src/views/partials/header.njk:10]: Decided to keep them inline for now, can't use `currentColor` with an `<object>` and so we would need 2 versions (white/grey) otherwise.

File: src/views/partials/nav.njk (hunk 6 above)
Comment (hamishtaplin) [reply src/views/partials/nav.njk]: Changed how this works so it's much cleaner syntax now

**Review by dandwal — DISMISSED** [review]
Comments have been addressed. Also looks great on Mobile, nice one.

**Review by dandwal — APPROVED** [review]
